sub-packages/bionemo-scbenchmark/src/bionemo/scbenchmark/common.py

```diff
@@ -0,0 +1,641 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Common components shared across the benchmarking framework.
+
+This module contains shared dataclasses and utility functions to avoid
+circular imports between modules. It provides the core data structures
+and measurement utilities used throughout the benchmarking framework.
+"""
+
+import gc
+import json
+import multiprocessing as mp
+import os
+import platform
+import subprocess
+import time
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import psutil
+
+
+@dataclass
+class BenchmarkResult:
+    """Results from benchmarking a dataloader.
+
+    This class stores all metrics and metadata about a dataloader benchmark run,
+    including timing, memory usage, throughput, configuration parameters, and
+    the raw data for detailed analysis.
+
+    Attributes:
+        name: Name/description of the benchmark
+        disk_size_mb: Size of the dataset files on disk in MB
+        setup_time_seconds: Time spent setting up the dataloader
+        warmup_time_seconds: Time spent in warmup phase
+        total_iteration_time_seconds: Time spent iterating through data (excluding warmup)
+        average_batch_time_seconds: Average time per batch
+        total_batches: Number of batches processed (excluding warmup)
+        total_samples: Number of samples processed (excluding warmup)
+        samples_per_second: Throughput in samples per second (excluding warmup)
+        batches_per_second: Throughput in batches per second (excluding warmup)
+        peak_memory_mb: Peak memory usage during benchmark
+        average_memory_mb: Average memory usage during benchmark
+        errors: List of error messages
+
+        # Instantiation metrics
+        instantiation_time_seconds: TOTAL time (dataset + dataloader)
+        dataset_instantiation_time_seconds: Time to load/create dataset only
+        dataloader_instantiation_time_seconds: Time to wrap dataset in dataloader only
+        peak_memory_during_instantiation_mb: Peak memory usage during instantiation
+
+        # Configuration metadata
+        madvise_interval: Memory advice interval setting used
+        data_path: Path to dataset used for benchmarking
+        max_time_seconds: Maximum time limit set for the benchmark
+        shuffle: Whether the dataloader was shuffled
+    """
+
+    name: str
+    disk_size_mb: float
+    setup_time_seconds: float
+    warmup_time_seconds: float
+    total_iteration_time_seconds: float
+    average_batch_time_seconds: float
+    total_batches: int
+    total_samples: int
+    samples_per_second: float
+    batches_per_second: float
+    peak_memory_mb: float
+    average_memory_mb: float
+    errors: List[str] = None
```

Comment:
can do this with defaultfactory too
```suggestion
    errors: List[str] =  field(default_factory=list)
```

Reply:
removed errors